hw/mcu/dialog/da1469x/syscfg.yml

```diff
@@ -454,6 +454,11 @@ syscfg.defs:
         restrictions:
             - '!GPADC_BATTERY'
 
+    WATCHDOG_DISABLE_ON_INIT:
+        description: >
+          'Enable watchdog on init'
+        value: 1
```

Comment:
Name **WATCHDOG_DISABLE_ON_INIT** is not really consistent with description. One can wander which one is true.
Dialog freezing watchdog is quite unique future (usually when watchdog is enabled it's impossible to stop it till reset).
I would name it rather more explicit like WATCHDOG_FREEZE_ON_INIT but that's just me (and I will not insist).
But changing description to something like 'Freez watchdog on init, watchdog will be unfrozen by hal_watchdog_enable()' could be more descriptive.

Reply:
I'd also prefix this option with `MCU_` as it will be more consistent with other syscfg options that are specific to particular MCU rather than a generic ones

Reply:
Yeah I kind of went back and forth with the name as well.  So some watchdogs are auto-started on reset.  Some are not.  Some are like doomsday devices and you can't freeze them.  This Dialog one is kind of an odd bird.  The implementation we currently have freezes the watchdog during init.  So to keep the default behavior of the driver, I added the DISABLE_ON_INIT and made it the default (=1). 

It didn't matter to me whether we were freezing it or not on init, what mattered to me was that this was the behavior that was coded into the init function and that people are currently used to.  I am quite happy to just remove the freeze altogether.

Reply:
The background reason of why I am doing any of this.... So the watchdog is enabled coming out of the ROM bootloader (either by reset or otherwise).  The mcuboot bootloader was freezing it and leaving us in a state where the watchdog was not running for the entiretly of the bootloader.  During production, we are soldering batteries on to the PCBs and the intermittent voltage connection during soldering was allowing chips to boot and then they get hung with a power fluctuation.  The only way to fix these was to somehow manage to attach via JTAG or disconnect the battery or let the battery run out.

Leaving the watchdog running, it could recover itself.

Reply:
Even if we re-enabled the watchdog, there is a window where a power fluctuation could hang us inbetween the _init and the _enable of the watchdog.

Reply:
Updated both and force pushed change.